docs/connect.html

```diff
@@ -543,6 +543,67 @@ <h6>ACL requirements</h6>
+
+    <p>To make plugins compatible with <code>SERVICE_LOAD</code>, it is necessary to add <a href="https://docs.oracle.com/javase/8/docs/api/java/util/ServiceLoader.html">ServiceLoader</a> manifests to your source code, which should then be packaged in the release artifact. Manifests are resource files in <code>META-INF/services/</code> named after their superclass type, and contain a list of fully-qualified subclass names, one on each line. Manifests should be located in the same module as the classes they refer to, not in the corresponding test resources.</p>
+
+    <p>In order for a plugin to be compatible, it must appear as a line in a manifest corresponding to the plugin superclass it extends. If a single plugin implements multiple plugin interfaces, then it should appear in each corresponding manifest. If you have no classes for a certain type of plugin, you do not need to include a manifest file for that type. If you have classes which should not be visible as plugins, they should be marked abstract. The following types are expected to have manifests:</p>
+
+    <ul>
+        <li><code>org.apache.kafka.connect.sink.SinkConnector</code></li>
+        <li><code>org.apache.kafka.connect.source.SourceConnector</code></li>
+        <li><code>org.apache.kafka.connect.storage.Converter</code></li>
+        <li><code>org.apache.kafka.connect.storage.HeaderConverter</code></li>
+        <li><code>org.apache.kafka.connect.transforms.Transformation</code></li>
+        <li><code>org.apache.kafka.connect.transforms.predicates.Predicate</code></li>
+        <li><code>org.apache.kafka.common.config.provider.ConfigProvider</code></li>
+        <li><code>org.apache.kafka.connect.rest.ConnectRestExtension</code></li>
+        <li><code>org.apache.kafka.connect.connector.policy.ConnectorClientConfigOverridePolicy</code></li>
+    </ul>
+
+    <p>For example, if you only have one connector with the fully-qualified name <code>com.example.MySinkConnector</code>, then only one manifest file must be added to resources in <code>META-INF/services/org.apache.kafka.connect.sink.SinkConnector</code>, and the contents should be similar to the following:</p>
+
+    <pre class="brush: resource;">
+# license header or comment
+com.example.MySinkConnector</pre>
+
+    <p>You should then verify that your manifests are correct. either with the <a href="#connect_plugindiscovery_compatibility">manual verification steps</a> or with unit tests. If you aren't using the <code>EmbeddedConnectCluster</code> you can add a test suite that starts up an <code>EmbeddedConnectCluster</code> and does nothing with it. If you are using the <code>EmbeddedConnectCluster</code>, then ensure you are using version 3.6 or later of the connect-runtime main and test artifacts. If you have a test that passes using the default <code>HYBRID_FAIL</code> strategy, then your migration is complete. You can then release the plugin normally, and operators can upgrade to the compatible version.</p>
```

Comment:
Nit: comma instead of period after "You should then verify that your manifests are correct"

Reply:
Also, same thought RE explicit mention of the `EmbeddedConnectCluster` class. The instructions on starting a worker in a testing environment with `HYBRID_FAIL` mode are great, we should just be a tad more general about how we describe that kind of setup. This is also useful since some users have their own integration testing environments which don't use the `EmbeddedConnectCluster` class but can still use the `HYBRID_FAIL` testing strategy.